fl33t/models/mixins.py

```diff
@@ -48,6 +81,17 @@ def fleets(self, offset=None, limit=None):
 
 class OneFleetMixin:  # pylint: disable=too-few-public-methods
     """For models with a parent fleet"""
+
+    _fleet = None
+
+    @property
     def fleet(self):
         """Return the parent fleet"""
-        return self._client.get_fleet(self.fleet_id)
+
+        if not self.fleet_id:
+            return None
+
+        if not self._fleet or self.fleet_id != self._fleet.fleet_id:
```

Comment:
Same comment here regarding the co-caching of `fleet_id` and `_fleet.fleet_id`.

Reply:
Waiting on the agreed solution regarding the `build` property so that it can be applied here too.